docker/upload.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/python3
+import subprocess
+import sys
+import os
+import time
+import json
+
+images = ["ubuntu", "graphiteapp/graphite-statsd", "g4s8/artipie-base"]
+
+# Start artipie with preconfigured docker repo
+def start_artipie():
+    print("Starting artipie")
+    artipie_yml = """
+meta:
+  storage:
+    type: fs
+    path: /var/artipie/configs
+  layout: flat
+"""
+    my_docker = """
+repo:
+  type: docker
+  storage:
+    type: fs
+    path: /var/artipie/data
+  permissions:
+    "*":
+      - "*"
+"""
+    os.makedirs("./configs", exist_ok=True)
+    f = open("./artipie.yaml", "w+")
+    f.write(artipie_yml)
+    f.close()
+    f = open("./configs/my-docker.yaml", "w+")
+    f.write(my_docker)
+    f.close()
+    subprocess.run([
+        "bash", "-c",
+        "docker run -d --rm --name artipie -it -v $(pwd)/artipie.yaml:/etc/artipie.yml -v $(pwd):/var/artipie -p 8080:80 artipie/artipie:latest"
+    ])
+
+
+# Start docker registry
+def start_registry():
+    print("Starting docker registry")
+    subprocess.run(["docker",
+                    "run",
+                    "-d",
+                    "--rm",
+                    "-it",
+                    "-p", "5000:5000",
+                    "--name", "registry",
+                    "registry:2"])
+
+
+# Measure execution time for passed function
+def measure(func):
+    tick = time.time()
+    func()
+    tock = time.time()
+    return tock - tick
+
+
+# Perform benchmark
+def perform_benchmarks(images):
+    result = \
+        {"docker":
+            {
+                "single-upload": {
+                    "artipie": {
+                        "images": []
+                    },
+                    "docker-registry": {
+                        "images": []
+                    }
+                }
+            }
+        }
+    single = result["docker"]["single-upload"]
+    for image in images:
+        registry_push = ["docker", "push", f"localhost:5000/{image}"]
```

Comment:
@Sammers21 any docker command execution may be delayed if docker daemon is not initialized. I'd suggest to call `docker info >> /dev/null` before any command to make sure that daemon was started.

Reply:
@g4s8, we call `docker run hello-world` on every instance. It's enough.